vicodyn/include/cocaine/gateway/vicodyn.hpp

```diff
@@ -76,6 +77,7 @@ class vicodyn_t : public api::gateway_t{
     dynamic_t::object_t locator_extra;
     api::gateway_ptr wrapped_gateway;
     vicodyn::peers_t peers;
+    std::shared_ptr<vicodyn::darkmetrics_t> darkmetrics;
```

Comment:
Why is `darkmetrics` stored in `std::shared_ptr`?

Reply:
Because I use enable_shared_for_this to prolong life of darkmetrics.

Reply:
Why is it needed?